engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -890,4 +880,14 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
     runTransaction(action)
   }
+
+  private def materializeWorkflowDescriptor(id: WorkflowId, workflowSourceFiles: WorkflowSourceFiles)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
```

Comment:
Just to be on the same page:
```scala
class WorkflowManagerActor(config: Config, materializeWfActor: ActorRef) {
...

// WMA created WorkflowActor, and say WorkflowActor also needs the `materializeWfActor` 
val wfActor = WorkflowActor.props(wfDescriptor, materializeWfActor)

//where the constructor of WorkflowActor is changed to also receive a reference to materializeWfActor

}
```

And for other singletons needing to materialize the WorkflowDescriptor (like `WorkflowMetadataBuilder`), does it still use `CromwellServer.materializeWfActor` ? Or is there a way to inject the actor ref to other Scala objects?